modules/geo/src/main/java/org/opensearch/geo/algorithm/PolygonGenerator.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.geo.algorithm;
+
+import org.apache.commons.collections4.CollectionUtils;
```

Comment:
Let's avoid adding this new dependency! There is an OpenSearch [CollectionUtils](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/common/util/CollectionUtils.java) class we can use (if it is missing something let's add it).

Reply:
What is the reason of not adding new dependency? I can see other modules/libs are already using this dependency.

Reply:
Example: https://github.com/opensearch-project/OpenSearch/blob/688c348ee1acceed5f4f6598111db0231fef6bed/plugins/ingest-attachment/build.gradle#L83 

I can go ahead and remove it also. I don't have a strong reason on either of the side

Reply:
removed the extra dependency.